mantidproject/mantid
# Support for SANSILLParameterScan for D16 in DrILL

**Description of work.**

This PR adds the support for `SANSILLParameterScan` algorithm for D16 instrument in DrILL interface.

**To test:**

With data in `UnitTest/ILL/D16/`:

* Add the data directory to your search directories
* Open DrILL interface
* Select D16 instrument and the acquisition mode "Sample scan"
* Fill the table as follow:

| SampleRuns | AbsorberRuns | ContainerRuns | OutputWorkspace | OutputJoinedWorkspace | CustomOptions |
|------------------:|:--------------------|:---------------------|:------------------------|:----------------------------------|:---------------------|
|023583:023585|                       |                          |                        out|                                joined|                           |

* Process the row (click on `Process all` button at the bottom right of the window)
* Check for the presence of `out` and `joined` workspaces
* Consult the history to see the call to `SANSILLParameterScan`

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
